app/models/manageiq/providers/terraform_enterprise/automation_manager/orchestration_stack/status.rb

```diff
@@ -0,0 +1,52 @@
+class ManageIQ::Providers::TerraformEnterprise::AutomationManager::OrchestrationStack::Status < ::OrchestrationStack::Status
+  def initialize(status)
+    self.status = status
+    self.reason = self.class.run_state_to_description(status)
+  end
+
+  def succeeded?
+    status.downcase == "planned_and_finished"
+  end
+
+  def failed?
+    %w[policy_soft_failed errored].include?(status.downcase)
+  end
+
+  def canceled?
+    %w[discarded canceled force_canceled].include?(status.downcase)
+  end
+
+  # Lookup table of run state descriptions
+  # https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#run-states
+  def self.run_state_to_description(status)
+    case status
+    when "pending"              then N_("Pending")
+    when "fetching"             then N_("The run is waiting for HCP Terraform to fetch the configuration from VCS.")
+    when "fetching_complete"    then N_("HCP Terraform has fetched the configuration from VCS and the run will continue.")
+    when "pre_plan_running"     then N_("The pre-plan phase of the run is in progress.")
+    when "pre_plan_completed"   then N_("The pre-plan phase of the run has completed.")
+    when "queuing"              then N_("HCP Terraform is queuing the run to start the planning phase.")
+    when "plan_queued"          then N_("HCP Terraform is waiting for its backend services to start the plan.")
```

Comment:
Does this "HCP Terraform" terminology change if it's the on-prem veresion?

Reply:
No, https://developer.hashicorp.com/terraform/enterprise/api-docs/run#run-states also states `HCP Terraform is waiting for its backend services to start the plan.`